### ISSUE TITLE
Fail to compile lyon_geom with version 0.7

### ISSUE THREAD

**Issue by buxx** [ggez/ggez#993]
**Describe the bug**

Hello team !

When I use the 0.7 ggez version, my program don't compile : 

```
➜  OpenCombat git:(ggez7) ✗ cargo build
   Compiling lyon_geom v0.17.5
   Compiling wayland-cursor v0.28.5
   Compiling wayland-protocols v0.28.5
   Compiling wayland-egl v0.28.5
   Compiling minimp3 v0.5.1
   Compiling glyph_brush_draw_cache v0.1.4
error[E0599]: no method named `x_range` found for reference `&Box2D<S, UnknownUnit>` in the current scope
   --> /home/bux/.cargo/registry/src/github.com-1ecc6299db9ec823/lyon_geom-0.17.5/src/line.rs:438:29
    |
438 |         self.clipped_x(clip.x_range())?.clipped_y(clip.y_range())
    |                             ^^^^^^^ method not found in `&Box2D<S, UnknownUnit>`

error[E0599]: no method named `y_range` found for reference `&Box2D<S, UnknownUnit>` in the current scope
   --> /home/bux/.cargo/registry/src/github.com-1ecc6299db9ec823/lyon_geom-0.17.5/src/line.rs:438:56
    |
438 |         self.clipped_x(clip.x_range())?.clipped_y(clip.y_range())
    |                                                        ^^^^^^^ method not found in `&Box2D<S, UnknownUnit>`

   Compiling zip v0.5.12
For more information about this error, try `rustc --explain E0599`.
error: could not compile `lyon_geom` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
```

You can find my game source code here : https://github.com/buxx/OpenCombat/tree/ggez7

**To Reproduce**

Try to `cargo build` the following project (branch `ggez7`) : https://github.com/buxx/OpenCombat/tree/ggez7

**Expected behavior**

No `lyon_geom` compile error

**Hardware and Software:**

    ggez = "0.7.0"

Ubuntu and cargo 1.56.0

**Comment by PSteinhaus** [comment]
Thanks for reporting!

It seems that lyon_geom is using two euclid functions that were not available in euclid 0.22.3, [but are available from 0.22.4 onwards](https://docs.rs/euclid/0.22.4/euclid/struct.Box2D.html#method.x_range).

Your Cargo.lock file though specifies euclid 0.22.3 to be used, so you get the error. Updating the lock file (or deleting it and recompiling) fixes your problem, as the newest version, euclid 0.22.6, is then automatically chosen and used.

It's late and I'd have to read a bit more into the Cargo documentation to actually be able to give you a proper explanation on why exactly this happened, but at least I can provide you with a quick solution right now.

**Comment by buxx** [comment]
@PSteinhaus  You got it ! I misunderstood the logs, thanks for your time !
